youtube_dl/extractor/twitch.py

```diff
@@ -138,13 +138,17 @@ def _prefer_source(self, formats):
         self._sort_formats(formats)
 
     def _download_base_gql(self, video_id, ops, note, fatal=True):
+        headers = {
+            'Content-Type': 'text/plain;charset=UTF-8',
+            'Client-ID': self._CLIENT_ID,
+        }
+        gql_auth = self._get_cookies('https://gql.twitch.tv').get('auth-token')
+        if gql_auth is not None:
+            headers['Authorization'] = 'OAuth ' + gql_auth.value
```

Comment:
this should be done once(in `_real_initialize`).

Reply:
Sorry, I don't understand what exactly you're asking for. I can't find any header setting function in `common.py`. Also, this header is only needed for the gql request. While it would be possible to set something like `self.gql_auth_header` in `_real_initialize`, this feels to me like adding gratuitous complexity.

Reply:
> I can't find any header setting function in `common.py`.

`_real_initialize` is used for extractor initialization, it's up to every extrator to setup it's own initialization.

> While it would be possible to set something like self.gql_auth_header in _real_initialize, this feels to me like adding gratuitous complexity.

there is no complexty, you will set the header once and reuse them in every subsequent request, instead of constructing the same headers over and over again for a large collections.

Reply:
Technically, cookie may change between requests so that moving `Authorization` calculation in `_real_initialize` may result in expired token.

Reply:
the assumption was based on the fact that the cookie is set programatically and not using `Set-Cookie` headers, but as it's undefined wheather the value can change or not, i guess it's better to set the cookie value for every request.